////////////////////////////////////////////////////////////////////////////////
/// @file         ProtocolMessage.proto
///
/// @author       Michael Catanzaro <michael.catanzaro@mst.edu>
///
/// @description  A ProtocolMessage encapsulates a ModuleMessage.
///
/// These source code files were created at Missouri University of Science and
/// Technology, and are intended for use in teaching or research. They may be
/// freely copied, modified, and redistributed as long as modified versions are
/// clearly marked as such and this notice is not removed. Neither the authors
/// nor Missouri S&T make any warranty, express or implied, nor assume any legal
/// responsibility for the accuracy, completeness, or usefulness of these files
/// or any information distributed with these files.
///
/// Suggested modifications or questions about these files can be directed to
/// Dr. Bruce McMillin, Department of Computer Science, Missouri University of
/// Science and Technology, Rolla, MO 65409 <ff@mst.edu>.
////////////////////////////////////////////////////////////////////////////////

import "ModuleMessage.proto";

package freedm.broker;

message ProtocolMessage
{
    enum Status
    {
        CREATED = 1;
        ACCEPTED = 2;
        BAD_REQUEST = 3;
    }

    required string source_uuid = 1;
    required string source_hostname = 2;
    required string source_port = 3;
    // FIXME times should be sent as ints, not as strings...
    required string send_time = 4;
    required string expire_time = 5;
    required uint32 sequence_no = 6;
    optional Status status = 7;
    optional int32 kill = 8;
    optional fixed64 hash = 9;

    optional ModuleMessage module_message = 10;
}
